src/components/Toggle/index.tsx

```diff
@@ -15,40 +15,49 @@ interface Props<T> {
 
 const Toggle = <T extends {}>(props: Props<T>) => {
   const { items, activeItem, onChange, variablesClassName, getLabel, getValue } = props;
+  const buttonWidth = 100 / items.length;
+  const calculateTranslation = items.indexOf(activeItem) * 100;
 
   return (
-    <div data-testid='toggle-element' className={classNames(styles.toggle, variablesClassName)}>
-      <div
-        className={classNames(styles['toggle-active-btn'], variablesClassName)}
-        data-active-item={getValue(activeItem) === getValue(items[0])}
-      />
-      {items.map(item => {
-        return (
-          <label
-            key={getLabel(item)}
-            htmlFor={getLabel(item)}
-            data-active={activeItem === item}
-            data-testid={`toggle-label-${getLabel(item)}`}
-          >
-            <Button
-              size='medium'
-              text={getLabel(item)}
-              category='neutral'
-              onClick={() => onChange(item)}
-              data-testid={`toggle-element-${getLabel(item)}`}
-            />
-            <input
-              readOnly
-              id={getLabel(item)}
-              type='radio'
-              name='toggle'
-              checked={activeItem === item}
-              value={getValue(item)}
-              data-testid={`toggle-input-${getLabel(item)}`}
-            />
-          </label>
-        );
-      })}
+    <div className={classNames(styles['toggle-wrapper'])}>
+      <div data-testid='toggle-element' className={classNames(styles.toggle, variablesClassName)}>
+        <div
+          className={classNames(styles['toggle-active-btn'], variablesClassName)}
+          style={{
+            width: `${buttonWidth}%`,
+            transform: `translateX(${calculateTranslation}%)`
```

Comment:
sets animation and width to be dynamic and to consider the number of items passed by the consumer.